cmake/module_switch.cmake

```diff
@@ -0,0 +1,42 @@
+#############################################################################
+# Copyright (c) 2019 Balabit
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+function (module_switch NAME HELP_STRING)
+  if (DEFINED ${NAME})
+    return()
+  endif()
+
+  if (DISABLE_ALL_MODULES)
```

Comment:
This way the `DISABLE_ALL_MODULES` is not visible on gui/tui (like `ccmake`). I would prefer to have all user option to be visible. What do you think ?

Reply:
Agreed. I will do so.